src/main/java/org/humancellatlas/ingest/migrations/MongoChangeLog.java

```diff
@@ -84,4 +84,12 @@ public void singletonSubmissionEnvelopeProject(MongoDatabase db) {
 
         db.getCollection("project").updateMany(filter, update);
     }
+
+    @ChangeSet(order = "2020-08-06", id = "Drop Alias Index on archiveEntity", author = "alexie.staffer@ebi.ac.uk")
+    public void dropAliasIndexOnArchiveEntity(MongoDatabase db) {
+        db.getCollection("archiveEntity").dropIndex("alias");
+        // If the collection does not exist this code will still succeed,
```

Comment:
Just found this comment interesting - there's a possibility that migrations can fail based on current state.

We should consider writing this migration log in a way that it can't fail when ran on a blank mongo instance. The very first changeset should set up the "initial conditions" i.e the collections we had, the indexes on them, etc. If all future changes come through this changelog (collection renames, index modifications, etc.) then we can be certain that a migration never fails (at least a DB-schema migration).

or tl:dr; the migration change log shouldn't be sensitive to DB state/schema, rather it should completely dictate DB state/schema

Reply:
I have not checked the whole migration set. Initially I thought that you wrote them like Rolando said. If not, then I totally agree with Rolando, we should consider to order/change it in a way that we could rebuild our DB from scratch with the help of this migration set.
And for testing we could have some special data set that we could add to a local or test db if needed to reproduce a case.